OWML.Patcher/BinaryPatcher.cs

```diff
@@ -0,0 +1,179 @@
+        // String that comes right before the bytes we want to patch.
+        private const string PatchZoneText = "Assets/Scenes/PostCreditScene.unity";
+        private const int PatchStartZoneOffset = 6;
+        private const int FileSizeStartIndex = 4;
+        private const int FileSizeEndIndex = FileSizeStartIndex + 4;
+        private const string FileName = "globalgamemanagers";
+        private const string BackupSuffix = ".bak";
+        private const int BuildSettingsStartAddressIndex = 0x2CC;
+        private const int BuildSettingsSizeIndex = 0x2D0;
+        private const int BlockAddressOffset = 0x1000;
+
+        public BinaryPatcher(IOwmlConfig owmlConfig, IModConsole writer)
+        {
+            _owmlConfig = owmlConfig;
+            _writer = writer;
+            _filePath = $"{_owmlConfig.DataPath}/{FileName}";
+        }
+
+        public void Patch()
+        {
+            if (!File.Exists(_filePath))
+            {
+                throw new FileNotFoundException(_filePath);
+            }
+
+            var fileBytes = File.ReadAllBytes(_filePath);
+
+            var buildSettingsStartIndex = BitConverter.ToInt32(fileBytes, BuildSettingsStartAddressIndex) + BlockAddressOffset;
+            var buildSettingsSize = BitConverter.ToInt32(fileBytes, BuildSettingsSizeIndex);
+            var buildSettingsEndIndex = buildSettingsStartIndex + buildSettingsSize;
+
+            var patchStartIndex = FindPatchStartIndex(fileBytes, buildSettingsStartIndex, buildSettingsEndIndex);
+            var isAlreadyPatched = FindExistingPatch(fileBytes, patchStartIndex, buildSettingsEndIndex);
+
+            if (isAlreadyPatched)
+            {
+                _writer.WriteLine("globalgamemanagers already patched.");
+                return;
+            }
+
+            BackupFile(_filePath);
+            var patchedBytes = CreatePatchedFileBytes(fileBytes, patchStartIndex);
+            File.WriteAllBytes(_filePath, patchedBytes);
+            _writer.WriteLine("Successfully patched globalgamemanagers.");
+        }
+
+        private int FindPatchStartIndex(byte[] fileBytes, int startIndex, int endIndex)
+        {
+            byte[] patchZoneBytes = Encoding.ASCII.GetBytes(PatchZoneText);
+            var patchZoneMatch = 0;
+            for (var i = startIndex; i < endIndex; i++)
+            {
+                var fileByte = fileBytes[i];
+                var patchZoneByte = patchZoneBytes[patchZoneMatch];
+                if (fileByte == patchZoneByte)
+                {
+                    patchZoneMatch++;
+                }
+                else
+                {
+                    patchZoneMatch = 0;
+                }
+                if (patchZoneMatch == patchZoneBytes.Length)
+                {
+                    return i + PatchStartZoneOffset;
+                }
+            }
+            throw new Exception("Could not find patch zone in globalgamemanagers. This probably means the VR patch needs to be updated.");
```

Comment:
It's recommended to use an exception which inherits from Exception and says something about what's up. Is there one that makes sense to use?

Reply:
I looked for it, but couldn't find anything that made sense. Open to suggestions. I used FileNotFoundException where it made sense.

Reply:
even better, then you can make your own 🥳 
PatchNotFoundException? 😁